internal/ethapi/api.go

```diff
@@ -997,13 +997,66 @@ func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNrOr
 }
 
 func DoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.BlockNumberOrHash, gasCap *big.Int) (hexutil.Uint64, error) {
-	// Retrieve the block to act as the gas ceiling
-	block, err := b.BlockByNumberOrHash(ctx, blockNrOrHash)
-	if err != nil {
-		return 0, err
+	// Binary search the gas requirement, as it may be higher than the amount used
```

Comment:
This is just a revert to the original function content right?

Reply:
Yes